docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,786 @@
+    models, particularly in FastAPI 
+    ([here](https://github.com/samuelcolvin/foxglove/blob/a4aaacf372178f345e5ff1d569ee8fd9d10746a4/foxglove/exceptions.py#L137-L149)
+    is one method I've used), we should provide utilities to generate these errors
+12. Improve the performance of `__eq__` on models
+13. Computed fields, these having been an idea for a long time in pydantic - we should get them right
+14. Model validation that avoids instances of subclasses leaking data (particularly important for FastAPI),
+    see [pydantic-core#155](https://github.com/samuelcolvin/pydantic-core/issues/155)
+
+## Removed Features :neutral_face:
+
+1. `__root__` custom root models are no longer necessary since validation on any supported data type is allowed
+   without a model
+2. `parse_file` and `parse_raw`, partially replaced with `.model_validate_json()`
+3. `TypeError` are no longer considered as validation errors, but rather as internal errors, this is to better
+   catch errors in argument names in function validators.
+4. Subclasses of builtin types like `str`, `bytes` and `int` are coerced to their parent builtin type,
+   this is a limitation of how pydantic-core converts these types to Rust types during validation, if you have a
+   specific need to keep the type, you can use wrap validators or custom type validation as described above
+5. [Settings Management](https://pydantic-docs.helpmanual.io/usage/settings/) ??? - I definitely don't want to
+   remove the functionality, but it's something of a historical curiosity that it lives within pydantic,
+   perhaps it should move to a separate package, perhaps installable alongside pydantic with 
+   `pip install pydantic[settings]`?
+6. The following `Config` properties will be  removed:
+   * `fields` - it's very old (it pre-dates `Field`), can be removed
+   * `allow_mutation` will be removed, instead `frozen` will be used
+   * `error_msg_templates`, it's not properly documented anyway, error messages can be customised with external logic if required
+   * `getter_dict` - pydantic-core has hardcoded `from_attributes` logic
+   * `json_loads` - again this is hard coded in pydantic-core
+   * `json_dumps` - possibly
+   * `json_encoders` - see the export "mode" discussion [above](#improvements-to-dumpingserializationexport)
+   * `underscore_attrs_are_private` we should just choose a sensible default
+   * `smart_union` - all unions are now "smart"
+
+## Features Remaining :neutral_face:
+
+The following features will remain (mostly) changed:
+
+* Generics
+* JSONSchema, internally this will need to change a lot, but hopefully the external interface will remain unchanged
+* `dataclass` support, again internals might change, but not the external interface
+* `validate_arguments`, might be renamed, but otherwise remain
+* hypothesis plugin, might be able to improve this as part of the general cleanup
```

Comment:
If Pydantic starts using Annotated and the annotated-types constraint objects, then I think most of the features currently provided by the plugin (i.e. handling constrained types) will just come for free 😁

Unsure if you'll need a Hypothesis plugin at all, at that point, though it's probably still useful if you want Hypothesis to e.g. not generate integers that don't fit in an int64... unless of course you auto-add the constraints for that!